internal/db/merge.go

```diff
@@ -43,28 +45,37 @@ func (db *db) handleMerges(ctx context.Context, merges events.Subscription[event
 				return
 			}
 			go func() {
-				err := db.executeMerge(ctx, merge)
+				// ensure only one merge per docID
+				queue.add(merge.DocID)
+				defer queue.done(merge.DocID)
+
+				var err error
+				// retry merge up to max txn retries
+				for i := 0; i < db.MaxTxnRetries(); i++ {
```

Comment:
thought: If you use the merge queue, I expect that a retry will never be needed.

Reply:
If a user updates a doc while a merge is in progress it could still happen.

Reply:
😅 very true

Reply:
> If a user updates a doc while a merge is in progress it could still happen.

I'd be super nice / reminder to document that here.